### ISSUE TITLE
dvc status <path spec> : Respect --cloud flag

### ISSUE THREAD

**Issue by johnyaku** [iterative/dvc#9336]
Bordering on a bug request, but I'll give you the benefit of the doubt and label this as a feature request, because it could simply be that my intuition is out of line with how dvc works.

When used locally (without the `--cloud` flag) `dvc status`  can operate on a path spec, but this fails confusingly when the `--cloud` flag is added.

Scenario: "folder" is tracked by dvc but has been deleted from the cache and/or the remote.
```
dvc status --cloud folder
```
.. will report "Cache and remote are in sync".

If `folder/` is tracked via `folder.dvc` then the correct status can be obtained via ...
```
dvc status --cloud folder.dvc
```

I feel that it would be more intuitive if `dvc status --cloud folder` worked out of the box.
Discord discussion: https://discordapp.com/channels/485586884165107732/1095524989870022667

**Comment by dberenbaum** [comment]
Thanks for the report @johnyaku! I guess since it works with `dvc status --cloud folder.dvc`, it's not blocking any scenario for you?

**Comment by johnyaku** [comment]
That's right, it's not blocking. Just took a couple of hours to figure out because the docs where not clear on this distinction between how `dvc status` operates locally versus remotely.

If `dvc status` is going to be deprecated then I'd understand if changing the behaviour is not a priority, but a sentence in the docs clarifying this might save folks some time :)
